parser/tcpinfo.go

```diff
@@ -96,15 +96,19 @@ func (p *TCPInfoParser) IsParsable(testName string, data []byte) (string, bool)
 	return "", false
 }
 
+// thinSnaps collects every 10th snapshot in orig. thinSnaps guarantees that the last snapshot is preserved.
 func thinSnaps(orig []snapshot.Snapshot) []snapshot.Snapshot {
 	n := len(orig)
+	if n == 0 {
```

Comment:
Regarding [this comment](https://github.com/m-lab/etl/pull/1105/commits/bebb994b3240cf86991dc51815159bd398a60921#r991514476): I am not 100% sure how thinSnaps behaves when the list has only one element.

Theoretically, the for loop should not do anything, as for `i=0` and `n=1` the condition `i < n-1` is false, but maybe it's worth checking?

To be super safe, one could change this check to `if n<=2`. I.e. the first and last element are always included (or the single element, if there is only one); and only if we have more than two elements, we apply the thinning logic.